collector.go

```diff
@@ -0,0 +1,104 @@
+package redigosrv
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+//RedigoCollector struct to access metrics
+type RedigoCollector struct {
+	publishTrafficSize prometheus.Counter
+	methodCalls        *prometheus.CounterVec
+	subscribeAmount    prometheus.Gauge
+	subscribeSuccess   prometheus.Counter
+	subscribeFailures  prometheus.Counter
+}
+
+// RedigoCollectorOptions struct to add custom name in metrics
+type RedigoCollectorOptions struct {
+	SplitBy string
+	Prefix  []string
+	Suffix  []string
+}
+
+const (
+	//PublishMetricMethodName has the name of the specific method
+	PublishMetricMethodName string = "Publish"
+	//SubscribeMetricMethodName has the name of the specific method
+	SubscribeMetricMethodName string = "Subscribe"
+)
+
+var redigoMetricLabels = []string{"method"}
+
+//RedigoCollectorDefaultOptions will return the instance of RedigoCollectorDefaultOptions with values default
+func RedigoCollectorDefaultOptions() RedigoCollectorOptions {
+	return RedigoCollectorOptions{
+		SplitBy: "_",
+		Prefix:  []string{"redigo"},
+		Suffix:  []string{"collector"},
+	}
+}
+
+/*
+	Output with default options:
+
+	Ex: fmt.Sprintf("%spublishTrafficSize%s", prefix, suffix)
+	Exit: redigo_publishTrafficSize_collector
+*/
+
+// NewRendigoCollector will return new instance of RedigoCollector with all metrics started
+func NewRendigoCollector(opts RedigoCollectorOptions) *RedigoCollector {
```

Comment:
```suggestion
func NewRedigoCollector(opts RedigoCollectorOptions) *RedigoCollector {
```

Rename  `Rendigo`  to `Redigo`

Reply:
Done!